pkg/apiaccesskeys/apiaccesskey.go

```diff
@@ -0,0 +1,103 @@
+package apiaccesskeys
+
+// APIAccessKeyBase represents a New Relic API access key.
+type APIAccessKeyBase struct {
+	ID    string `json:"id,omitempty"`
+	Key   string `json:"key,omitempty"`
+	Name  string `json:"name,omitempty"`
+	Notes string `json:"notes,omitempty"`
+	Type  string `json:"type,omitempty"`
+}
+
+// APIAccessIngestKey represents a New Relic API access ingest key.
+type APIAccessIngestKey struct {
+	APIAccessKeyBase
+	AccountID  int    `json:"accountId,omitempty"`
+	IngestType string `json:"ingestType,omitempty"`
+}
+
+// APIAccessUserKey represents a New Relic API access user key.
+type APIAccessUserKey struct {
+	APIAccessKeyBase
+	AccountID int    `json:"accountId,omitempty"`
+	UserID    string `json:"userId,omitempty"`
+}
+
+type apiAccessIngestKeyCreateResponse struct {
+	APIAccessKeys struct {
+		CreatedKeys []APIAccessIngestKey `json:"createdKeys"`
+	} `json:"apiAccessCreateKeys"`
+}
+
+type apiAccessIngestKeyDeleteResponse struct {
+	APIAccessKey struct {
+		DeletedKey []struct {
+			ID string `json:"id"`
+		} `json:"deletedKeys,omitempty"`
+		Errors []struct {
+			Message string `json:"message,omitempty"`
+			Type    string `json:"type,omitempty"`
+		}
+	} `json:"apiAccessDeleteKeys"`
+}
+
+const (
+	graphqlAPIAccessIngestKeyFields = `
+		id
+		key
+		name
+		notes
+		type
+		... on ApiAccessIngestKey {
+			ingestType
+			accountId
+		}
+	`
+
+	apiAccessIngestKeyCreateKeys = `mutation($keys: Int!){
+			apiAccessCreateKeys(keys: $keys) {createdKeys {` + graphqlAPIAccessIngestKeyFields + `}
+		}}`
+
+	apiAccessIngestKeyDeleteKey = `mutation($keys: Int!){
+			apiAccessDeleteKeys(keys: $keys) {
+				deletedKeys {
+					id
+				}
+				errors {
+					message
+					type
+				}
+		}}`
+)
+
+// CreateAPIAccessIngestKeyMutation creates a new ingest type API access key.
+func (a *APIAccessKeys) CreateAPIAccessIngestKeyMutation(keys APIAccessCreateIngestKeysInput) ([]APIAccessIngestKey, error) {
```

Comment:
This is great. Similar comment to earlier, we should consider that this mutation facilitates creating **ingest keys** _and_ **user keys**. So maybe the API looks something like this: 

```suggestion
func (a *APIAccessKeys) CreateAPIAccessKeysMutation(keys APIAccessCreateKeysInput) ([]APIAccessKey, error) {
```

This would facilitate creating both types of keys and make the client's API a 1:1 mapping to the GraphQL mutation signature 😎

Reply:
Ah I was hoping to make it easier for users to use specific methods named after the key type. Also, my motivation to work on this PR is for ingest keys only.

I can explore a consolidated approach but if it proves a bit too much for me, I will likely back out and implement these endpoints on my fork.